sycl/source/detail/device_binary_image.hpp

```diff
@@ -330,6 +332,10 @@ class CompressedRTDeviceBinaryImage : public RTDeviceBinaryImage {
 private:
   std::unique_ptr<char[]> m_DecompressedData;
   size_t m_ImageSize = 0;
+
+  // Flag to ensure decompression happens only once.
+  std::once_flag m_InitFlag;
+  bool m_IsCompressed = true;
```

Comment:
@aelovikov-intel Following up on our conversation offline:
I'm wondering if we really need an `atomic<bool>` here?
There's a race on `m_IsCompressed` when `Decompress()` is trying to change its value and `IsCompressed` is reading it. But, what's the worse that can happen? The thread will try to decompress again, but it is anyway protected using `std::call_once`. 

`IsCompressed()` is called on hot path, just before `urProgramCreate`, so, a regular `bool` would be more performant, since the race is benign. WDYT?

Reply:
I'm not a memory model expert, but I think without `atomic` compiler/HW might be allowed to re-order lines 730/732 in the .cpp above. If that's true and happens, then initializing thread can set `m_IsCompressed = false` and then yield to the scheduler without updating `Format`. If another thread sees the `false` and reads `Format`, the value there would be wrong.

Reply:
Theocratically, yes, compiler can reorder the lines.... I'll change it to `atomic<bool>` then